modules/sdk-coin-apt/src/lib/transaction/transferTransaction.ts

```diff
@@ -20,37 +22,81 @@ export class TransferTransaction extends Transaction {
     this._assetId = APTOS_COIN;
   }
 
+  private isValidPayload(payload: TransactionPayload): boolean {
+    return (
+      payload instanceof TransactionPayloadEntryFunction &&
+      payload.entryFunction.args.length === 2 &&
+      payload.entryFunction.type_args.length === 1 &&
+      payload.entryFunction.type_args[0].toString().length > 0
+    );
+  }
+
+  private parseRecipients(
+    addressEntryFunctionArg: EntryFunctionArgument,
+    amountEntryFunctionArg: EntryFunctionArgument
+  ): TransactionRecipient[] {
+    const addressBytes = addressEntryFunctionArg.bcsToBytes();
+    const amountBytes = amountEntryFunctionArg.bcsToBytes();
+    if (addressBytes.length > 32 || amountBytes.length > 8) {
+      const deserializedAddresses = utils.deserializeVectorOfAddresses(addressBytes);
+      const deserializedAmounts = utils.deserializeVectorOfAmounts(amountBytes);
+      if (deserializedAddresses.length !== deserializedAmounts.length) {
+        throw new InvalidTransactionError(
+          'Invalid payload entry function arguments : addresses and amounts length mismatch'
+        );
+      }
+      return deserializedAddresses.map((address, index) => ({
+        address: address.toString(),
+        amount: utils.getAmountFromPayloadArgs(deserializedAmounts[index].bcsToBytes()),
+      })) as TransactionRecipient[];
+    } else {
+      return [
+        {
+          address: addressEntryFunctionArg.toString(),
+          amount: utils.getAmountFromPayloadArgs(amountBytes),
+        },
+      ] as TransactionRecipient[];
+    }
+  }
+
   protected parseTransactionPayload(payload: TransactionPayload): void {
```

Comment:
Good refactoring!

let's follow [template-method](https://refactoring.guru/design-patterns/template-method) design pattern.
Summary:
1. Create a child class of TransferTransaction named BatchTransferTransaction.
2. Create a new TransactionType named `BatchSend` or reuse existing `Batch`.
3. return this TransactionType for COIN_BATCH_TRANSFER_FUNCTION from getTransactionTypeFromTransactionPayload()
4. instantiate BatchTransferTransaction() and override the protected method `parseRecipients`